File: langs/java/lib/src/codebot/Codebot.java
package codebot;

public class Codebot
{
    public static enum Direction { FORWARD, BACKWARD }
    public static enum Axis { X, Y }

    public static void init()
    {
        System.out.println("Init");
    }

    public static void step(Direction direction, Axis axis)
    {
        movement("Step", direction, axis);
    }

    public static void jump(Direction direction, Axis axis)
    {
        movement("Jump", direction, axis);
    }

    public static void openDoor(Direction direction, Axis axis)
    {
        movement("OpenDoor", direction, axis);
    }

    public static void pickUp()
    {
        System.out.println("PickUp");
    }

    public static void drop()
    {
        System.out.println("Drop");
    }

    public static void interact()
    {
        System.out.println("Interact");
    }

    private static void movement(String string, Direction direction, Axis axis)
    {
        if (axis == Axis.X)
        {
            switch (direction)
            {
                case FORWARD:
                    System.out.println(string + " Forward X");
                    break;

                case BACKWARD:
                    System.out.println(string + " Backward X");
                    break;
            }
        }
        else if (axis == Axis.Y)
        {
            switch (direction)
            {
                case FORWARD:
                    System.out.println(string + " Forward Y");
                    break;

                case BACKWARD:
                    System.out.println(string + " Backward Y");
                    break;
            }
        }
    }
}

File: langs/python/lib/codebot.py
import axis as axis_
import direction as direction_

def init():
    print("Init")

def step(direction, axis):
    movement("Step", direction, axis)

def jump(direction, axis):
    movement("Jump", direction, axis)

def openDoor(direction, axis):
    movement("openDoor", direction, axis)

def pickUp():
    print("PickUp");

def drop():
    print("Drop");

def interact():
    print("Interact");

def movement(string, direction, axis):
    if (axis == axis_.X):
        if (direction == direction_.FORWARD):
            print(string + " Forward X")
        elif (direction == direction_.BACKWARD):
            print(string + " Backward X")

    elif (axis == axis_.Y):
        if (direction == direction_.FORWARD):
            print(string + " Forward Y")
        elif (direction == direction_.BACKWARD):
            print(string + " Backward Y")
